src/api/app/models/user.rb

```diff
@@ -840,6 +840,15 @@ def mark_login!
     update_attributes(last_logged_in_at: Time.now, login_failure_count: 0)
   end
 
+  def pending_announcement
+    announcement = Announcement.last
+    announcement unless announcement.in?(announcements)
+  end
+
+  def acknowledge_announcement!(announcement)
```

Comment:
Why do you obscure this with a method around `CollectionProxy` assignment? People know `CollectionProxy` assignment. People have to look up what `User.acknowledge_announcement!` does.

Reply:
Good point:+1: We've updated it